keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -36,6 +38,52 @@ class SqlDeliveryConfigRepository(
 
   private val mapper = configuredObjectMapper()
 
+  override fun deleteByApplication(application: String) {
+    jooq.deleteFrom(DELIVERY_CONFIG)
+      .where(DELIVERY_CONFIG.APPLICATION.eq(application))
+      .execute()
+
+    jooq.deleteFrom(ENVIRONMENT)
+      .where(
+        notExists(jooq.select(DELIVERY_CONFIG.UID)
```

Comment:
Avoid using a sub-query for deletes, the lock implications aren't ideal. It would be preferable to first `select uid from delivery_config where application..`,  also select relevant environment ids, and delete everything based on those.